uint32[] fold_data
uint32[] test_data
uint32[] fold_sizes
uint32[] test_sizes
